tiledb/ml/readers/pytorch.py

```diff
@@ -87,17 +94,45 @@ def __iter__(self) -> Iterator[DataType]:
             iter_start = worker_id * per_worker
             iter_end = min(iter_start + per_worker, rows)
 
+        offsets = range(iter_start, iter_end, self.batch_size)
+
+        # Shuffle offsets in case we need batch shuffling
+        if self.batch_shuffle:
+            random.seed()
+            offsets = random.sample(  # type: ignore
```

Comment:
`batch_shuffle` is a misleading name for what this does; it doesn't shuffle, it (sub)samples.

Reply:
True but the final result is shuffling of the existing offsets. I will check if I can do it with numpy shuffle functionality.

Reply:
Ah indeed, with the given `k` passed to `sample()` it is equivalent to shuffling. Yeah better use `shuffle()` explicitly then for clarity.